lib/testing/src/lib/core/pages/form/formFields.ts

```diff
@@ -15,149 +15,140 @@
-        BrowserVisibility.waitUntilElementIsNotOnPage(this.formContent);
-        return this;
+    async noFormIsDisplayed(): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsNotVisible(this.formContent);
     }
 
-    checkFormIsDisplayed() {
-        BrowserVisibility.waitUntilElementIsVisible(this.formContent);
-        return this;
+    async checkFormIsDisplayed(): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.formContent);
     }
 
-    getNoFormMessage() {
+    async getNoFormMessage(): Promise<string> {
         return BrowserActions.getText(this.noFormMessage);
     }
 
-    getCompletedTaskNoFormMessage() {
+    async getCompletedTaskNoFormMessage(): Promise<string> {
         return BrowserActions.getText(this.completedTaskNoFormMessage);
     }
 
-    clickOnAttachFormButton() {
-        BrowserActions.click(this.attachFormButton);
-        return this;
+    async clickOnAttachFormButton(): Promise<void> {
+        await BrowserActions.click(this.attachFormButton);
     }
 
-    selectForm(formName) {
-        BrowserActions.click(this.selectFormDropDownArrow);
-        BrowserVisibility.waitUntilElementIsVisible(this.selectFormContent);
-        this.selectFormFromDropDown(formName);
-        return this;
+    async selectForm(formName): Promise<void> {
+        await BrowserActions.click(this.selectFormDropDownArrow);
+        await BrowserVisibility.waitUntilElementIsVisible(this.selectFormContent);
+        await this.selectFormFromDropDown(formName);
     }
 
-    selectFormFromDropDown(formName) {
+    async selectFormFromDropDown(formName): Promise<void> {
         const formNameElement = element(by.cssContainingText('span', formName));
-        BrowserActions.click(formNameElement);
+        await BrowserActions.click(formNameElement);
     }
 
-    checkWidgetIsReadOnlyMode(fieldId) {
+    async checkWidgetIsReadOnlyMode(fieldId): Promise<ElementFinder> {
         const widget = element(by.css(`adf-form-field div[id='field-${fieldId}-container']`));
         const widgetReadOnly = widget.element(by.css('div[class*="adf-readonly"]'));
-        BrowserVisibility.waitUntilElementIsVisible(widgetReadOnly);
+        await BrowserVisibility.waitUntilElementIsVisible(widgetReadOnly);
         return widgetReadOnly;
     }
 
-    completeForm() {
-        BrowserActions.click(this.completeButton);
+    async completeForm(): Promise<void> {
+        await BrowserActions.click(this.completeButton);
     }
 
-    setValueInInputById(fieldId, value) {
+    async setValueInInputById(fieldId, value): Promise<void> {
         const input = element(by.id(fieldId));
-        BrowserActions.clearSendKeys(input, value);
-        return this;
+        await BrowserActions.clearSendKeys(input, value);
     }
 
-    isCompleteFormButtonDisabled() {
-        BrowserVisibility.waitUntilElementIsVisible(this.completeButton);
+    async isCompleteFormButtonDisabled(): Promise<string> {
```

Comment:
according to the name, return type should be Promise<boolean>

Reply:
not sure I got is a promise

Reply:
sorry, should have escaped some chars, wanted to say that return type should be Promise of void, not string, according to the method name